lib/buf.c

```diff
@@ -109,13 +121,7 @@ M0_INTERNAL int m0_buf_copy_aligned(struct m0_buf *dst,
 				    unsigned       shift)
 {
 	M0_PRE(dst->b_nob == 0 && dst->b_addr == NULL);
-
-	M0_ALLOC_ARR_ALIGNED(dst->b_addr, src->b_nob, shift);
-	if (dst->b_addr == NULL)
-		return M0_ERR(-ENOMEM);
-	dst->b_nob = src->b_nob;
-	memcpy(dst->b_addr, src->b_addr, src->b_nob);
-	return 0;
+	return m0_buf_new_aligned(dst, src->b_addr, src->b_nob, shift);
```

Comment:
We can define the src as const:
```
const struct m0_buf *src,
```

This is similar to "m0_buf_copy()".

And do you want to simplify the "m0_buf_copy()" to share common code?

Reply:
Good spot about `const`, will do. About `m0_buf_copy()` - not sure...